tests/test-rest-posts-controller.php

```diff
@@ -455,6 +457,20 @@ public function test_get_items_private_filter_query_var() {
 		$this->assertEquals( $draft_id, $data[0]['id'] );
 	}
 
+	public function test_get_items_invalid_per_page() {
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array( 'per_page' => -1 ) );
+		$response = $this->server->dispatch( $request );
+		$this->assertErrorResponse( 'rest_invalid_param', $response, 400 );
+	}
+
+	public function test_get_items_invalid_filter_posts_per_page() {
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_query_params( array( 'filter' => array( 'posts_per_page' => -1 ) ) );
+		$response = $this->server->dispatch( $request );
+		$this->assertErrorResponse( 'rest_invalid_param', $response, 400 );
+	}
```

Comment:
I don't know a good way to error for that outside of validating the filter param, which has been decided against in past conversations.  When #2699 is merged, I would be +1 on changing the test to check if the response has greater than 1 and less than 10 posts, as that is the default `per_page amount`. Alternatively you could check the handler for the response to see what the request variables were and see if `per_page` was `10`.